Update responses docs (#6722)
RasaHQ/rasa: docs/docs/glossary.mdx

  Incoming messages are processed by a sequence of components called a pipeline. A component can perform tasks ranging from entity extraction to intent classification to pre-processing.

## [Response / Template / Utterance](./responses.mdx)

>   A message template that is used to respond to a user. This can include text, buttons, images, and other content.

Reviewer: ```suggestion
  A message that an assistant sends to a user. This can include text, buttons, images, and other content.
```
